src/quacc/atoms/skzcam.py

```diff
@@ -6,35 +6,173 @@
+) -> tuple[Atoms, Atoms, int, NDArray, NDArray]:
+    """
+    Read the file containing the periodic slab and adsorbate (geometry optimized) and return the key information needed to create an embedded cluster in ChemShell.
+
+    Parameters
+    ----------
+    adsorbate_slab_file
+        The path to the file containing the adsorbate molecule on the surface slab. It can be in any format that ASE can read.
+    adsorbate_idx
+        The indices of the atoms that make up the adsorbate molecule.
+    slab_center_idx
+        The the adsorbate_slab_file, this gives the indices of the atoms that are at the 'center' of the slab right beneath the adsorbate.
+
+    Returns
+    -------
+    adsorbate
+        The Atoms object of the adsorbate molecule.
+    slab
+        The Atoms object of the surface slab.
+    center_position
+        The position of the center of the cluster.
+    list[float, float, float]
+        The vector from the center of the slab to the center of mass of the adsorbate.
+    """
+
+    # Get the necessary information for the cluster from a provided slab file (in any format that ASE can read)
+    adsorbate_slab = read(adsorbate_slab_file)
+
+    # Find indices (within adsorbate_slab) of the slab
+    slab_idx = [x for x in list(range(len(adsorbate_slab))) if x not in adsorbate_idx]
+
+    # Create slab from adsorbate_slab
+    slab = adsorbate_slab[slab_idx].copy()
+
+    # Find index of the first center atom of the slab as listed in slab_center_idx
+    slab_first_atom_idx = [
+        index for index, x in enumerate(slab_idx) if x == slab_center_idx[0]
+    ][0]
+
+    # Get the centre of the cluster from the atom indices
+    slab_center_position = np.zeros(3)
+    for atom_idx in slab_center_idx:
+        slab_center_position += adsorbate_slab.get_positions()[atom_idx]
+
+    slab_center_position = slab_center_position / len(slab_center_idx)
+
+    adsorbate = adsorbate_slab[adsorbate_idx].copy()
+
+    # Get the relative distance of the adsorbate from the first center atom of the slab as defined in the slab_center_idx
+    adsorbate_com = adsorbate.get_center_of_mass()
+    adsorbate_vector_from_slab = (
+        adsorbate[0].position - adsorbate_slab[slab_center_idx[0]].position
+    )
+
+    # Add the height of the adsorbate from the slab along the z-direction relative to the first center atom of the slab as defined in the slab_center_idx
+    adsorbate_com_z_disp = (
+        adsorbate_com[2] - adsorbate_slab[slab_center_idx[0]].position[2]
+    )
+    center_position = (
+        np.array([0.0, 0.0, adsorbate_com_z_disp])
+        + slab_center_position
+        - adsorbate_slab[slab_center_idx[0]].position
+    )
+
+    return (
+        adsorbate,
+        slab,
+        slab_first_atom_idx,
+        center_position,
+        adsorbate_vector_from_slab,
+    )
+
+
+@requires(chemshell_module, "ChemShell is not installed")
+def run_chemshell(
```

Comment:
Maybe we should rename this as `create_embedded_cluster` or something more descriptive since it's not really a general utility to run chemshell.